Special footer

sarahvc opened the pull request:
Added a special footer option based on this design:
https://uxframework.pearson.com/c/special-application-header#footer

davodey commented:
@sarahvc I noticed there is no bg color set to the footer, so when it goes to responsive mode, you see the content underneath.

<img width="621" alt="screen shot 2018-07-24 at 3 58 49 pm" src="https://user-images.githubusercontent.com/5407196/43170592-be83b1ce-8f5a-11e8-9248-fd69b0c221d0.png">

davodey commented:
@sarahvc Would it be possible to display multiple footers on the demo page?  At least for some of the more prominent options, like the logo and maybe dark mode?  I feel this would help out devs and designers both.

davodey commented on src/components/Footer/Footer.js:
```diff
@@ -18,23 +18,44 @@ const Footer = ({ copyrightText, links, anchorTarget, light, singlePageStick })
+    );
 }
 
 export default Footer;
```
We need to add a propType for the logo.  Probably something like: `logo: PropTypes.bool`

sarahvc replied:
Added the protoTypes and defaultProp.

davodey commented on src/components/Footer/Footer.js:
```diff
@@ -18,23 +18,44 @@ const Footer = ({ copyrightText, links, anchorTarget, light, singlePageStick })
-        {renderCopy()}
-      </footer>
-  );
+  const lightCheck = light ? ' pe-footer--light':'';
```
This looks good.  If you wanted to trim it down you could by adding a style object.  If your interested, check it out: [https://reactjs.org/docs/dom-elements.html#style](https://reactjs.org/docs/dom-elements.html#style)

sarahvc commented:
@davodey Thanks Dave! I've pushed another commit, in which I 1) updated the logo version footer after confirmed with UXF designers. 2) added logo into the the protoType, and added default value for logo, 3)Added demo of different options in the demo page, 4)Removed demo footer's singlePageStick props to false.

More about 4): the singlePageStick props is intended for pages that content do not exceed one screen. But the demo page actually exceeds one screen when squeezed to a smaller screen size. So it's not suitable to set this props to true. I was trying to find a solution that we could make the footer work for scenario like this. All methods that I found involves doing something to the content. I think that would be beyond the scope for the footer component. So I left it as it is and updated the props to prompt consumers to check for all screen size before setting this props to true. Let me know if this makes sense.

Thank you!
